src/python/pants/init/target_roots_calculator.py

```diff
@@ -146,126 +86,43 @@ def create(cls, options, session, symbol_table, build_root=None, exclude_pattern
     logger.debug('spec_roots are: %s', spec_roots)
     logger.debug('changed_request is: %s', changed_request)
     logger.debug('owned_files are: %s', owned_files)
-    scm = get_scm()
-    change_calculator = ChangeCalculator(scheduler=session, symbol_table=symbol_table, scm=scm) if scm else None
-    owner_calculator = OwnerCalculator(scheduler=session, symbol_table=symbol_table) if owned_files else None
     targets_specified = sum(1 for item
                          in (changed_request.is_actionable(), owned_files, spec_roots)
                          if item)
 
     if targets_specified > 1:
-      # We've been provided a more than one of: a change request, an owner request, or spec roots.
+      # We've been provided more than one of: a change request, an owner request, or spec roots.
       raise InvalidSpecConstraint(
         'Multiple target selection methods provided. Please use only one of '
         '--changed-*, --owner-of, or target specs'
       )
 
-    if change_calculator and changed_request.is_actionable():
+    if changed_request.is_actionable():
+      scm = get_scm()
+      if not scm:
+        raise InvalidSpecConstraint(
+          'The --changed-* options are not available without a recognized SCM (usually git).'
+        )
+      changed_files = cls.changed_files(
+          scm,
+          changes_since=changed_request.changes_since,
+          diffspec=changed_request.diffspec)
       # We've been provided no spec roots (e.g. `./pants list`) AND a changed request. Compute
       # alternate target roots.
-      changed_addresses = change_calculator.changed_target_addresses(changed_request)
+      request = OwnersRequest(sources=tuple(changed_files),
+                              include_dependees=str(changed_request.include_dependees))
+      changed_addresses, = session.product_request(BuildFileAddresses, [request])
```

Comment:
Not for this change, but it seems like a `session.product_request_single` should exist to avoid the awkwardness of the more explicit `changed_addresses = ...; assert len(changed_addresses) == 1; changed_address = changed_addresses[0]` and the visual subtlety of a trailing comma singleton tuple assert/extract.